### PR TITLE
Correcting argument to code2color

### PR DESCRIPTION
**Pull request by sharad**
lang="-l ${lang#.}"

to

lang="-l${lang#.}"

as when calling with lang=c

code2color $PPID ${in_file:+"$in_file"} "$lang" "$2"

shell see "-l c" so " c" is received by code2color

### CODE REVIEW

File: lesspipe.sh.in [hunk 1]
```diff
@@ -669,7 +669,7 @@ isfinal() {
       msg "append :. or :<filetype> to activate syntax highlighting"
     else
       lang=${3#$sep}
-      lang="-l ${lang#.}"
```
Comment (Konfekt) [inline lesspipe.sh.in:672]: Could you make the same change to the file `lesspipe.sh`?

**Comment by Konfekt** [comment]
Hasn't the change to `lesspipe.sh.in` become dispensable by the trimming of whitespace in `code2color`?

File: lesspipe.sh.in (hunk 1 above)
Comment (sharad) [reply lesspipe.sh.in:672]: Yes did it.

Thanks for reply.

**Comment by sharad** [comment]
> Hasn't the change to lesspipe.sh.in become dispensable by the trimming of whitespace in code2color?


Still each component should show correct behaviour independenly.

**Comment by Konfekt** [comment]
The perl doc,`Getopt::Std`says:

> Switches which take an argument don't care whether there is a space between the switch and the argument.

Therefore, why is this white space trimming necessary?

**Comment by sharad** [comment]
But in practice, when I was using 

less cfilepath/fcile.c:c


found the no color is appearing as argument is passed like "code2coder" .. "-l c" ..
Inside this perl script it take argument value as " c" so not able to find revelent regular expression
which finally resulted into no color.

**Comment by sharad** [comment]
I was trying to create a guix package for lesspipe

https://github.com/sharad/guix/blob/master/lotus/packages/utils.scm#L35

**Comment by sharad** [comment]
> The perl doc,`Getopt::Std`says:
> 
> > Switches which take an argument don't care whether there is a space between the switch and the argument.
> 
> Therefore, why is this white space trimming necessary?

**Comment by sharad** [comment]
By mistake I closed this

**Comment by sharad** [comment]
> > The perl doc,`Getopt::Std`says:
> > > Switches which take an argument don't care whether there is a space between the switch and the argument.
> > 
> > 
> > Therefore, why is this white space trimming necessary?

Yes what perl doc,`Getopt::Std`says: 
is correct

but here issue how shell is passing argument

code2color $PPID ${in_file:+"$in_file"} "$lang" "$2"

which finally passed like

"code2color" "PPIDNUM" "-ic" "-l c" "filepath/file.c"

$0= code2color
$1=PPIDNUM
$2="-ic"
$3="-l c"
$4="filepath/file.c"

If same command is used with interactive bash that it will like
code2color PPIDNUM -ic -l c filepath/file.c

or

"code2color" "PPIDNUM" "-ic" "-l" "c" "filepath/file.c"

$0= code2color
$1=PPIDNUM
$2="-ic"
$3="-l"
$4="c"
$5="filepath/file.c"

we can directly test code2color behaviour directly on bash by passign `-l c' 

one time "-l" "c"
another time with "-l c"

**Comment by Konfekt** [comment]
I merge this. I understand why `lesspipe.sh` had to be adapted, but still not why `c2c_trim` is necessary.

**Comment by Konfekt** [comment]
Thank you!

**Comment by sharad** [comment]
> I merge this. I understand why `lesspipe.sh` had to be adapted, but still not why `c2c_trim` is necessary.



Thanks

c2c_trim is not necessary, will remove it.

**Pull request merged**